theme/plugins/client-plugin.js

```diff
@@ -28,6 +32,12 @@ class ClientPlugin extends PluginInterface
         customClasses.objects['people_town_1'] = Npc1;
     }
 
+    defineEngineDriver(props)
```

Comment:
> I'm thinking that we should have a single driver pass through instead of checking the props.gameConfigEngine.
I really want to avoid the condition 'phaser' condition, so we could create a driver mapper like adding the drivers to the custom classes config:

`
props.gameManager.config.client.customClasses.engineDrivers = {
     phaser: PhaserDriver
};
`

I'm ok with this but I'm still having issues understanding what "customClasses" are in this scope.

`this.gameEngineDriver = new config.client.engineDrivers[this.initialGameData.clientEngineDriver]();`

I'm also ok with this code, but I think is better to pass the map of engines and the selected/configured engine driver  to the GameEngine's constructor instead of doing the instantiation in the GameManager. Because the GameManager don't need or have to know how is the implementation of the GameEngine, it just have to pass to the GameEngine all it needs to work and is his job to do the instantiation of the EngineDriver (after checking if the parameters exists and every is ok). (is more a implementation of the single responsabily rule).

Reply:
Let's schedule a call for next week